### PR TITLE
Internal: add styles command [EDS-257]

### PR DESCRIPTION
**Pull request by ronros-elementor**

### CODE REVIEW

File: includes/base/widget-base.php [hunk 1]
```diff
@@ -649,8 +649,6 @@ public function render_content() {
 
 			$this->print_widget_css();
 
-			// get_name
```
Comment (Omerisra6) [inline includes/base/widget-base.php:652]: maybe move to another PR

File: includes/base/atomic-widget-base.php [hunk 2]
```diff
@@ -34,6 +34,18 @@ final public function get_initial_config() {
...
+	final public function get_raw_data( $with_html_content = false ) {
+		$raw_data = parent::get_raw_data( $with_html_content );
+
+		$data = $this->get_data();
+
+		if ( ! empty( $data['styles'] ) ) {
+			$raw_data['styles'] = $data['styles'];
+		}
+
+		return $raw_data;
```
Comment (Omerisra6) [inline includes/base/atomic-widget-base.php:46]: Maybe create a property in the constructor that will save the styles attribute and then us it in this method

File: assets/dev/js/editor/document/elements/commands/styles.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
+/**
+ * @typedef {import('../../../container/container')} Container
```
Comment (Omerisra6) [inline assets/dev/js/editor/document/elements/commands/styles.js:2]: Do we need it?

File: assets/dev/js/editor/document/elements/commands/styles.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+export class Styles extends $e.modules.editor.document.CommandHistoryDebounceBase {
+	/**
+	 * Function getSubTitle().
+	 *
+	 * Get sub title by container.
+	 *
+	 * @return {string} sub title
+	 */
+	static getSubTitle() {
+		return 'Styles';
```
Comment (Omerisra6) [inline assets/dev/js/editor/document/elements/commands/styles.js:13]: translates?

File: assets/dev/js/editor/document/elements/commands/styles.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+	static restore( historyItem, isRedo ) {
+		const data = historyItem.get( 'data' );
+
+		historyItem.get( 'containers' ).forEach( ( /* Container */ container ) => {
+			const changes = data.changes[ container.id ];
+
+			$e.run( 'document/elements/styles', {
+				container,
+				styles: isRedo ? changes.new : changes.old,
+			} );
```
Comment (Omerisra6) [inline assets/dev/js/editor/document/elements/commands/styles.js:33]: Why don't we need the external option?

File: assets/dev/js/editor/document/elements/commands/styles.js (hunk 3 above)
Comment (ronros-elementor) [reply assets/dev/js/editor/document/elements/commands/styles.js:2]: Yes it's for the JSDoc comment for `addToHistory` (line 37)

```
	 /**
	 * Function addToHistory().
	 *
	 * @param {Container} container
	 * @param {{}}        newStyles
	 * @param {{}}        oldStyles
	 */
```

File: assets/dev/js/editor/document/elements/commands/styles.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+					old: oldStyles,
+					new: newStyles,
+				},
+			},
+			historyItem = {
+				containers: [ container ],
+				data: { changes },
+				type: 'change',
+				restore: Styles.restore,
+			};
```
Comment (Omerisra6) [inline assets/dev/js/editor/document/elements/commands/styles.js:56]:
```suggestion
		const changes = {
				[ container.id ]: {
					old: oldStyles,
					new: newStyles,
				},
			};
			const historyItem = {
				containers: [ container ],
				data: { changes },
				type: 'change',
				restore: Styles.restore,
			};
```

File: assets/dev/js/editor/document/elements/commands/styles.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+		containers.forEach( ( container ) => {
+			container = container.lookup();
+
+			const newStyles = styles,
+				oldStyles = container.model.get( 'styles' );
+
+			// Clear old oldValues.
+			container.oldValues = {};
+
+			// Set oldValues, For each setting is about to change save setting value.
```
Comment (Omerisra6) [inline assets/dev/js/editor/document/elements/commands/styles.js:90]: change comment

File: assets/dev/js/editor/document/elements/commands/styles.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+			// If history active, add history transaction with old and new styles.
+			if ( this.isHistoryActive() ) {
+				this.addToHistory( container, newStyles, container.oldValues );
+			}
+
+			container.model.set( 'styles', styles );
+		} );
+	}
+}
```
Comment (Omerisra6) [inline assets/dev/js/editor/document/elements/commands/styles.js:103]: is there any way to test it?
